actionpack/lib/action_controller/test_case.rb

```diff
@@ -593,7 +607,14 @@ def process(action, http_method = 'GET', *args)
           @request.env['RAW_POST_DATA'] = args.shift
         end
 
-        parameters, session, flash = args
+        if args[0] && args[0].keys.any? { |k| REQUEST_KWARGS.include?(k) }
+          parameters = args[0][:params]
+          session =    args[0][:session]
+          flash =      args[0][:flash]
+        else
+          parameters, session, flash = args
```

Comment:
We should deprecate this path or we will not be able to use kwargs for real here in the future.